dataProcessing.py

```diff
@@ -70,32 +70,72 @@ def fill_data(nhits, hitList, db):
 
 
 yeast_name_map = {}
+syst_name_map = {}
+std_name_map = {}
+# Fill name mapping dictionaries
+def fill_name_maps():
+    global yeast_name_map, syst_name_map, std_name_map
+    with open("yeast_names_ref.txt", "r") as f:
+        for line in f:
+            words = line.split()
+            yeast_name = words[0]
+            syst_name = words[1]
+            std_name = ''
+            new_name = syst_name
+            description = syst_name+' (hypothetical protein)'
+            if len(words)==3:
+                std_name = words[2]
+                new_name = std_name
+                description = std_name
+                syst_name_map[std_name] = syst_name
+            yeast_name_map[yeast_name] = new_name, description
+            std_name_map[syst_name] = std_name
+
 # Fix ORF name from yeast database
 def yeast_name_fixed(name):
     global yeast_name_map
     if len(yeast_name_map)==0:
-        fill_yeast_name_map()
-    return yeast_name_map[name]
-
-# Fill yeast name mapping dictionary
-def fill_yeast_name_map():
-    global yeast_name_map
-    with open("Saccharomyces_cerevisiae_S288c.fas", "r") as f:
-        for line in f:
-            if line[0]!='>':
-                continue
-            words = line.split()[0:-3]
-            old_name = (words[0])[1:]
-            new_name = old_name
-            description = new_name+' '+' '.join(words[1:])
-            if len(words)==2:
-                new_name = words[1]
-                description = new_name
-            else:
-                if words[1]=='hypothetical':
-                    new_name = words[3]
-                    description = new_name+' (hypothetical protein)'
-            yeast_name_map[old_name] = new_name, description
+        fill_name_maps()
+    return yeast_name_map[name] if name in yeast_name_map else name
+
+# Check if name is systematic/standard name
+def is_systematic_name(name):
+    global std_name_map
+    if len(std_name_map)==0:
```

Comment:
The repetition of this pattern is a bit ugly. I'd be tempted to refactor all the maps into a single object using properties to hide the lazy loading. So you can do e.g.
```python
def is_systematic_name(name):
    return name in name_maps.std
```

Defined using something like
```python
class NameMaps(object):
    def __init__(self):
        self._std = {}
    @property
    def std(self):
        if not self._std:
            self._fill_maps()
        return self._std

name_maps = NameMaps()
```
With similar bits for the other maps, and the fill maps function becoming a method.

Reply:
Yep. Done.